core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaStoreMessageHandler.java

```diff
@@ -34,6 +34,25 @@ public KafkaStoreMessageHandler(KafkaSchemaRegistry schemaRegistry,
     this.store = store;
   }
 
+  /**
+   * Invoked before every new K,V pair written to the store
+   *
+   * @param key   Key associated with the data
+   * @param value Data written to the store
+   */
+  public boolean validateUpdate(SchemaRegistryKey key, SchemaRegistryValue value) {
+    if (key.getKeyType() == SchemaRegistryKeyType.SCHEMA) {
+      SchemaValue schemaObj = (SchemaValue) value;
+      if (schemaObj != null) {
+        if (schemaRegistry.guidToSchemaKey.containsKey(schemaObj.getId())) {
```

Comment:
I think this would need to check the actual schemas as well. We have the notion of the same id for a given schema even across subjects.